Gems/ROS2/Code/Include/ROS2/Lidar/LidarRaycasterBus.h

```diff
@@ -135,7 +120,7 @@ namespace ROS2
         //! @param lidarTransform Current transform from global to lidar reference frame.
         //! @param flags Used to request different kinds of data returned by raycast query
         //! @return Results of the raycast in the requested form including 3D space coordinates and/or ranges.
-        virtual RaycastResult PerformRaycast(const AZ::Transform& lidarTransform) = 0;
+        virtual AZ::Outcome<RaycastResults, const char*> PerformRaycast(const AZ::Transform& lidarTransform) = 0;
```

Comment:
This is intentional as the errors are simply error messages and all of them are c-style string literals. Those are statically allocated and don't need to be dealocated manually. I will however put an additional comment about this in documentation so that it is clear to the user.

Reply:
Ok it resolved than.